tests/middleware.test.js

```diff
@@ -265,8 +265,8 @@ describe('validate body middleware', () => {
 
       await new Promise((resolve, reject) => {
         socket.on('connect', () => reject('Should not be able to connect'));
-        socket.on('error', error => {
-          if (error === 'Authentication error') resolve();
+        socket.on('connect_error', error => {
+          if (error.message === 'Authentication error') resolve();
           else reject(new Error('Wrong error: ' + error));
```

Comment:
Sounds like the old version had `error` as a string, but now it's an error object? If so line 270 should also be updated to include the error message rather than the whole error. Same below on line 289.

Does the error object have other properties? I wonder if testing something like `error.code` (if something like that exists) might be more reliable than the error message, which can sometimes be updated without treating it as a breaking change (semver major version increment).

Reply:
It looks like additional details can be attached to the `Error` object in [socketAuth.js](https://github.com/Emberex/rapid/blob/master/lib/middleware/socketAuth.js#L22) according to the [docs](https://socket.io/docs/v4/middlewares/#handling-middleware-error).

Reply:
Does it provide types, like can you hover over `error` to see it's type or go-to-definition for `.on`? Using `.message` is fine, I just thought I'd check.

Reply:
Hovering over the `error` param shows me the `any` type, same with `.on`, which didn't take me to a definition. Console logging the error did showed me this:

```
Error: Authentication error
          at Socket.onpacket (/Users/minohsuh/Documents/projects/rapid/node_modules/socket.io-client/build/cjs/socket.js:485:29)
          at Manager.Object.<anonymous>.Emitter.emit (/Users/minohsuh/Documents/projects/rapid/node_modules/@socket.io/component-emitter/index.js:143:20)
          at /Users/minohsuh/Documents/projects/rapid/node_modules/socket.io-client/build/cjs/manager.js:237:18
          at processTicksAndRejections (node:internal/process/task_queues:82:21) {
        data: undefined
}
```

Reply:
Oh fun, `any`. Well we could test whether it's `instanceof Error`, or you could add a `getErrorMessage` function. But maybe we just want to get the package working for now.